lib/chef/provider/package/macospkg.rb

```diff
@@ -0,0 +1,80 @@
+#
+# Author:: Joshua Timberman (<jtimberman@chef.io>)
+# Author:: William Theaker (<william.theaker+chef@gusto.com>)
+# Copyright:: Copyright (c) Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../provider/package"
+require_relative "package"
+
+class Chef
+  class Resource
+    class MacosPkg < Chef::Provider::Package
+      provides:macos_pkg
+
+      def load_current_resource
+        @current_resource = Chef::Resource::Package.new(new_resource.name)
+        current_resource.package_name(new_resource.package_name)
+        current_resource.version(get_current_version)
+        logger.trace("#{new_resource} current package version: #{current_resource.version}") if current_resource.version
+
+        download_pkg if new_resource.source
```

Comment:
we shouldn't do this at load_current_resource time - don't mess with the system  until we're in the phase where we chagne the system.